src/components/Employee/Deductions/DeductionsForm/ChildSupportForm.tsx

```diff
@@ -0,0 +1,262 @@
+  useEffect(() => {
+    setValue('caseNumber', null)
+    setValue('orderNumber', null)
+    setValue('remittanceNumber', null)
+  }, [watchedStateAgency, setValue])
+
+  const onChildSupportSubmit: SubmitHandler<ChildSupportPayload> = async data => {
+    const childSupport = {
+      state: data.state,
+      paymentPeriod: data.paymentPeriod,
+      fipsCode: data.fipsCode,
+      caseNumber: data.caseNumber,
+      orderNumber: data.orderNumber,
+      remittanceNumber: data.remittanceNumber,
+    }
+
+    await baseSubmitHandler(data, async payload => {
+      const requestBody = {
+        active: true,
+        amount: payload.amount,
+        description: `Child Support - ${childSupport.caseNumber}`, // child support description follows prefix type + case number convention
+        courtOrdered: true,
+        garnishmentType: 'child_support' as GarnishmentType,
+        times: null,
+        deductAsPercentage: true, // child support must always deduct as percentage up to a pay period maximum limit and is recurring until cancelled
+        payPeriodMaximum: payload.payPeriodMaximum,
+        recurring: true,
+        childSupport,
+      }
+
+      if (!deduction) {
+        const { garnishment: createDeductionResponse } = await createDeduction({
+          request: {
+            employeeId,
+            requestBody,
+          },
+        })
+        onEvent(componentEvents.EMPLOYEE_DEDUCTION_CREATED, createDeductionResponse)
+      } else {
+        const { garnishment: updateDeductionResponse } = await updateDeduction({
+          request: {
+            garnishmentId: deduction.uuid,
+            requestBody: {
+              ...requestBody,
+              version: deduction.version as string,
+            },
+          },
+        })
+        onEvent(componentEvents.EMPLOYEE_DEDUCTION_UPDATED, updateDeductionResponse)
+      }
+    })
+  }
+
+  const isManualPaymentRequired = selectedAgency?.manualPaymentRequired
+
+  return (
+    <FormProvider {...childSupportFormMethods}>
+      <Form onSubmit={childSupportFormMethods.handleSubmit(onChildSupportSubmit)}>
+        <Flex flexDirection="column" gap={32}>
+          <SelectField
+            name="state"
+            label={t('agency')}
+            description={t('agencyHelperText')}
+            options={stateAgencies}
+            onChange={handleStateAgencySelect}
+            isRequired
+          />
+          {watchedStateAgency && (
+            <>
+              <SelectField
+                name="fipsCode"
+                label={t('county')}
+                description={t('countyHelperText')}
+                options={counties}
+                isRequired
+              />
+              {/* render required inputs for respective agency, e.g. OH requires case number + order number */}
+              {requiredSelectedAgencyAttributes.map(({ name, label, description }) => (
+                <TextInputField
```

Comment:
Not sure if this is the right thing to do, but i'm wondering how often this re-renders and if we should memoize it for perf reasons?

Reply:
yea not much if any impact since the position in the tree/linked list doesnt change for the most prevalent rendered option (CSE Number), FWIW there are only 2 states (out of 51 options) which would either append 1 additional input, or wipe the LL/tree and replace with a different input altogether